src/Orleans/Configuration/ClusterConfiguration.cs

```diff
@@ -419,58 +419,55 @@ public string ToString(string siloName)
 
         internal static async Task<IPAddress> ResolveIPAddress(string addrOrHost, byte[] subnet, AddressFamily family)
         {
-            var loopback = (family == AddressFamily.InterNetwork) ? IPAddress.Loopback : IPAddress.IPv6Loopback;
+            var loopback = family == AddressFamily.InterNetwork ? IPAddress.Loopback : IPAddress.IPv6Loopback;
+
+            // IF the address is an empty string, default to the local machine
+            if (string.IsNullOrEmpty(addrOrHost))
+            {
+                addrOrHost = Dns.GetHostName();
+            }
 
-            if (addrOrHost.Equals("loopback", StringComparison.OrdinalIgnoreCase) ||
-                addrOrHost.Equals("localhost", StringComparison.OrdinalIgnoreCase) ||
-                addrOrHost.Equals("127.0.0.1", StringComparison.OrdinalIgnoreCase))
+            // Fix StreamFilteringTests_SMS tests
+            if (addrOrHost.Equals("loopback", StringComparison.OrdinalIgnoreCase))
```

Comment:
> So we lost 'localhost'?
The function `Dns.GetHostAddressesAsync(addrOrHost)` returns loopback IP in case  `addrOrHost == "localhost"`. It works if OS contains loopback interface (all current OS).